extensions/amp-pinterest/0.1/pin-widget.js

```diff
@@ -0,0 +1,194 @@
+import {assertHttpsUrl} from '../../../src/url';
+
+import {Util} from './util';
+
+const makePin = () => {
+
+  // make an embedded pin widget
+  // pinId will be inferred from pinUrl
+
+  let pinId = '';
+  try {
+    pinId = pinUrl.split('/pin/')[1].split('/')[0];
+  } catch (err) {
+    // fail silently
+    return;
+  }
+
+  const width = this.element.getAttribute('data-width');
+
+  const structure = make({'span': {}});
+
+  const renderPin = function(r) {
```

Comment:
I just updated this file. I copied and pasted it into it's own file for safe keeping, but decided to go ahead and integrate it into the new structure. So it's not as pretty as the others, but is updated and works now.